### PR TITLE
Chapter3

### PR DESCRIPTION
**Pull request by Perceval62**
### Solutions for Chapter 3

cc @vrom911 @chshersh

Kindly asking for a review of chapter three.
I tried to do the tasks as I saw them : )

### CODE REVIEW

File: src/Chapter3.hs [hunk 1]
```diff
@@ -971,6 +1088,33 @@ implement the following functions:
...
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend day = 
+  if day == Saturday || day == Sunday
+    then True
+    else False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day = 
+  if day == Sunday
```
Comment (vrom911) [inline src/Chapter3.hs:1109]: You could also use the instance of `Bounded` in here:

```suggestion
    if day == maxBound then minBound
```

This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂

File: src/Chapter3.hs [hunk 2]
```diff
@@ -971,6 +1088,33 @@ implement the following functions:
...
+    then Monday 
+    else succ day 
+
+daysToParty :: DayOfWeek -> Int
+daysToParty day
+  | day == Friday = 0
+  | day < Friday = length (enumFromTo day Friday) - 1
+  | day > Friday = (2 - length (enumFromTo Friday day)) + 6
```
Comment (vrom911) [inline src/Chapter3.hs:1117]: This could be done in a single line without all these cases with the use of `mod` function 🙂

File: src/Chapter3.hs [hunk 3]
```diff
@@ -1006,6 +1150,45 @@ Implement data types and typeclasses, describing such a battle between two
...
+data Beast = Beast
+  { beastHealth :: Hlt
+  , beastAttack :: Atk
+  }
+
+flee :: Beast -> Beast
+flee beast = beast {beastHealth = 0} 
+
+class Attack target where
+  attack :: Either Hero Beast -> target -> target
```
Comment (vrom911) [inline src/Chapter3.hs:1179]: Instead of `Either Hero Beast` it is better to use the polymorphic value, which should be of the class `Fighter`

**Review by vrom911 — APPROVED** [review]
Great Chapter! Very creative constructions you've built here 👷🏼 Very well done 👏🏼

**Comment by Perceval62** [comment]
Thanks for the feedback, I took a bit more time to complete this one to really get it. :D .
I will try to learn from it as best I can for the last chapter !

**Pull request closed without merging**